python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+-----------
+MatrixTable
+-----------
+
+A :class:`.MatrixTable` is a distributed two-dimensional dataset consisting of
+four components: a two-dimensional matrix where each entry is indexed by row
+key(s) and column key(s), a corresponding rows table that stores all of the row
+fields which are constant for every column in the dataset, a corresponding
+columns table that stores all of the column fields that are constant for every
+row in the dataset, and a set of global fields that are constant for every entry
+in the dataset.
+
+Unlike a :class:`.Table` which has two field groups (row fields and global
+fields), a matrix table has four field groups: global fields, row fields, column
+fields, entry fields.
+
+In addition, there are different operations on the matrix for each field group.
+For instance, :class:`.Table` has :meth:`.Table.select` and
+:meth:`.Table.select_globals`, and :class:`.MatrixTable` has
+:meth:`.MatrixTable.select_rows`, :meth:`.MatrixTable.select_cols`,
+:meth:`.MatrixTable.select_entries`, and :meth:`.MatrixTable.select_globals`.
+
+It is possible to represent matrix data by coordinate in a table , storing one
+record per entry of the matrix. However, the :class:`.MatrixTable` represents
+this data far more efficiently and exposes natural interfaces for computing on
+it.
+
+The :meth:`.MatrixTable.rows` and :meth:`.MatrixTable.cols` methods return the
+row and column fields as separate tables. The :meth:`.MatrixTable.entries`
+method returns the matrix as a table in coordinate form -- use this object with
+caution.
+
+Keys
+====
+
+Matrix tables have keys just as tables do. However, instead of one key, matrix
+tables have two keys: a row key and a column key. Row fields are indexed by the
+row key, column fields are indexed by the column key, and entry fields are
+indexed by the row key and the column key. The key structs can be accessed with
+:attr:`.MatrixTable.row_key` and :attr:`.MatrixTable.col_key`. It is possible to
+change the key with :meth:`.MatrixTable.key_rows_by` and
+:meth:`.MatrixTable.key_cols_by`.
+
+Note that changing the row key, however, may be an expensive operation.
```

Comment:
not sure the however is necessary. I bias towards giving folks justifications for our statements. I would say
> Due to the data representation of a matrix table, changing a row key is often an expensive operation.